### ISSUE TITLE
Brave loses all sessions and some saved passwords on close/restart

### ISSUE THREAD

**Issue by zanlib0** [brave/brave-browser#34622]
## Description 

Brave loses all sessions and some saved passwords on close/restart. This happens intermittently, however is pretty much guaranteed when restarting the computer or logging out.

## Steps to Reproduce

1. Open any website and login with selected “remember me”
2. Close Brave (kill with killall brave after closing)
3. Reopen and revisit the site

## Actual result:

Sometimes the session will persist, sometimes not. Oftentimes the saved passwords will be gone as well.

## Expected result:

The sessions don't get deleted.

I don’t have any extensions installed, and I don’t have the setting to clear cookies and sessions on close enabled.

## Reproduces how often: 

Intermittent issue, but very frequent.

## Brave version (brave://version info)
Brave	1.60.118 Chromium: 119.0.6045.163 (Official Build) (64-bit) 
Revision	b0599b790dbdfe28428202afe2b54758a044ccfc
OS	Linux

## Version/Channel Information:

- Can you reproduce this issue with the current release? 
- Can you reproduce this issue with the beta channel? 
- Can you reproduce this issue with the nightly channel?

Haven't checked, however I use the latest version from AUR and it's been happening since 1.59. 

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? No.
- Does the issue resolve itself when disabling Brave Rewards? I have them permanently disabled.
- Is the issue reproducible on the latest version of Chrome? I don't know, but it does not happen on Vivaldi.

## Miscellaneous Information:

OS: EndeavourOS Linux x86_64
Kernel: 6.6.2-arch1-1
WM: i3

I'm not even sure how to debug this issue. If there are any additional steps you'd like me to try, or any additional information I can provide, please suggest them. This issue makes Brave unusable for me, I've switched to Vivaldi since and it's not been a problem there.

**Comment by mdcowan** [comment]
I've also had this issue.

**Comment by voedev** [comment]
Yes, the problem is recurring with me as well. For example, gitlab.com. After rebooting my pc, I get logout from my account

**Comment by 1muen** [comment]
Didnt happned to me so far. Is your Brave Browser saving the passwords in the keyring of your OS? One easy way to tell is if you are greeted by an prompt asking you to unlock your Default Keyring with your password.

**Comment by MauroArgentino** [comment]
Apparently they haven't solved it yet. Exactly the same thing happens to me.
When I close the browser, all my sessions are closed, or at least the one I use the most, which are Google's.
I have been able to notice that Google, YouTube, etc. sessions are closed. when I have little space left on the disk, since the Brave browser, once opened, begins to eat resources from the hard drive and takes up 8 GB of space, leaving me with 25 megabytes or 0 megabytes on the disk. And when I close Brave to free up space, and then open it again, I encounter this. It does not store open sessions.

**Comment by Suvansarkar** [comment]
This issue is still persistent and it is getting quite annoying at this point. I have to relogin into all of my accounts 5-8 times a week at this point because of this. The passwords are being deleted (I don't know how) from the keyring of my OS where the passwords are actually stored as @1muen quoted earlier.

> Didnt happned to me so far. Is your Brave Browser saving the passwords in the keyring of your OS? One easy way to tell is if you are greeted by an prompt asking you to unlock your Default Keyring with your password.

Yes, it is infact storing all of its passwords in the keyring of my OS. I do put in my keyring password to open brave on a fresh startup.

**Comment by sanosis** [comment]
I get this behavior when I fail to unlock the keyring or brave can't connect to the keyring for whatever reason…

**Comment by diogo-vaz** [comment]
Is this after a year not yet solved?

**Comment by Suvansarkar** [comment]
@Splitrox I have switched to a different browser right now but I think installing a password storage software like KDE_Wallet or Gnome keyring should help solve the issue.

**Comment by zaneschepke** [comment]
Installing libgnome-keyring (optional dep) fixed this for me.

**Comment by carljustineoyales** [comment]
Still an issue to this day

**Comment by juanmad** [comment]
I had this happen and the issue was the Cookies file, somehow got corrupted so I just deleted it and it all started working again, no issues, I am on linux mint so path is /home/yourusername/.config/BraveSoftware/Brave-Browser/Default

Find the file called Cookies and delete it

I found only found this because I was trying to back up my brave folder and I got a I/O error when trying to copy the folder and it pointed to that Cookies file specifically so I just deleted it and then brave will create a new one and it will start working again.

**Comment by bsclifton** [comment]
Root cause for this may be https://github.com/brave/brave-browser/issues/50411 - please subscribe there for updates. I'm going to close this as a duplicate

One troubleshooting step for sure is to either delete the cookies file (using site data icon) or login to the problematic website and then immediately log out (to hopefully flush the cookie values). Next login should be good to go
